flink-runtime/src/test/java/org/apache/flink/runtime/persistence/TestingLongStateHandleHelper.java

```diff
@@ -106,17 +114,26 @@ public long getValue() {
 
         @Override
         public void discardState() {
-            preDiscardCallback.run(value);
-
-            numberOfDiscardCalls++;
+            try {
+                preDiscardCallback.run(numberOfDiscardCalls);
+                numberOfSuccessfulDiscardCalls++;
+            } catch (Exception e) {
```

Comment:
Out of curiosity: What do we gain out of it to make the callback throw a checked `Exception` and translating it here into a `RuntimeException` instead of just letting `PrecDiscardCallback` throw the `RuntimeException` rightaway? 🤔

Reply:
Just being explicit that this is primarily for the cases where we throw an exception. Also I'm using checked exceptions in the test suite.